src/main/java/org/openstreetmap/atlas/proto/adapters/ProtoAtlasMetaDataAdapter.java

```diff
@@ -0,0 +1,108 @@
+                protoAtlasMetaData.getRelationNumber());
+
+        final boolean original = protoAtlasMetaData.getOriginal();
+        final String codeVersion = protoAtlasMetaData.getCodeVersion();
+        final String dataVersion = protoAtlasMetaData.getDataVersion();
+        final String country = protoAtlasMetaData.getCountry();
+        final String shardName = protoAtlasMetaData.getShardName();
+        final Map<String, String> tags = PROTOTAG_LIST_CONVERTER
+                .convert(protoAtlasMetaData.getTagsList());
+
+        final AtlasMetaData atlasMetaData = new AtlasMetaData(atlasSize, original, codeVersion,
+                dataVersion, country, shardName, tags);
+
+        return atlasMetaData;
+    }
+
+    @Override
+    public byte[] serialize(final ProtoSerializable serializable)
+    {
+        if (!(serializable instanceof AtlasMetaData))
+        {
+            throw new CoreException(
+                    "Invalid ProtoSerializable type was provided to {}: cannot serialize {}",
+                    this.getClass().getName(), serializable.getClass().getName());
+        }
+        final AtlasMetaData atlasMetaData = (AtlasMetaData) serializable;
+
+        final ProtoAtlasMetaData.Builder builder = ProtoAtlasMetaData.newBuilder();
+        builder.setEdgeNumber(atlasMetaData.getSize().getEdgeNumber());
+        builder.setNodeNumber(atlasMetaData.getSize().getNodeNumber());
+        builder.setAreaNumber(atlasMetaData.getSize().getAreaNumber());
+        builder.setLineNumber(atlasMetaData.getSize().getLineNumber());
+        builder.setPointNumber(atlasMetaData.getSize().getPointNumber());
+        builder.setRelationNumber(atlasMetaData.getSize().getRelationNumber());
+        builder.setOriginal(atlasMetaData.isOriginal());
+
+        final Optional<String> codeVersionOptional = atlasMetaData.getCodeVersion();
+        if (codeVersionOptional.isPresent())
+        {
+            builder.setCodeVersion(codeVersionOptional.get());
+        }
+
+        final Optional<String> dataVersionOptional = atlasMetaData.getDataVersion();
+        if (dataVersionOptional.isPresent())
+        {
+            builder.setDataVersion(dataVersionOptional.get());
+        }
+
+        final Optional<String> countryOptional = atlasMetaData.getCountry();
+        if (countryOptional.isPresent())
+        {
+            builder.setCountry(countryOptional.get());
+        }
+
+        final Optional<String> shardNameOptional = atlasMetaData.getShardName();
+        if (shardNameOptional.isPresent())
+        {
+            builder.setShardName(shardNameOptional.get());
+        }
+
+        final Map<String, String> tags = atlasMetaData.getTags();
+        builder.addAllTags(ProtoAtlasMetaDataAdapter.PROTOTAG_LIST_CONVERTER.backwardConvert(tags));
+
+        final ProtoAtlasMetaData protoMetaData = builder.build();
+        return protoMetaData.toByteArray();
```

Comment:
how about

```
return builder.build().toByteArray();
```

Reply:
I had that intermediate variable there just to make it very clear to the reader what we are returning.

I'm actually going to build on your suggestion (no pun intended) by refactoring the name of 'builder' to be a little more descriptive, something like 'protoMetaDataBuilder'.

Best of both worlds that way!